coriolis/tests/osmorphing/test_windows.py

```diff
@@ -0,0 +1,893 @@
+            self.morphing_tools._compile_static_ip_conf_from_registry(
+                mock.sentinel.key_name)
+        self.conn.exec_ps_command.assert_has_calls([
+            mock.call(
+                "(((Get-ChildItem -Path '%s').Name | Select-String -Pattern "
+                "'[^\\\\]+$').Matches).Value" % interfaces_reg_path),
+            mock.call("(Get-ItemProperty -Path '%s\\interface1').EnableDHCP" %
+                      interfaces_reg_path),
+        ])
+
+    def test_check_ips_info(self):
+        nics_info = [
+            {'ip_addresses': [mock.sentinel.ip_address]}
+        ]
+        ips_info = [
+            {'ip_address': mock.sentinel.ip_address}
+        ]
+        self.morphing_tools._check_ips_info(nics_info, ips_info)
+
+    def test_check_ips_info_different_ips(self):
+        nics_info = [
+            {'ip_addresses': [
+                mock.sentinel.ip_address, mock.sentinel.ip_address2]}
+        ]
+        ips_info = [
+            {'ip_address': mock.sentinel.ip_address}
+        ]
+
+        self.assertRaises(exception.OSMorphingException,
+                          self.morphing_tools._check_ips_info,
+                          nics_info, ips_info)
+
+    @mock.patch.object(
+        windows.BaseWindowsMorphingTools, '_get_cbslinit_scripts_dir'
+    )
+    @mock.patch.object(windows.utils, 'write_winrm_file')
+    @mock.patch.object(windows.json, 'dumps')
+    @mock.patch('base64.b64encode')
+    def test__write_static_ip_script(self, mock_b64encode, mock_json_dumps,
+                                     mock_write_winrm_file,
+                                     mock_get_cbslinit_scripts_dir):
+        self.morphing_tools._write_static_ip_script(
+            mock.sentinel.base_dir, mock.sentinel.nics_info,
+            mock.sentinel.ips_info)
+
+        contents = windows.STATIC_IP_SCRIPT_TEMPLATE % {
+            'nics_info': mock_b64encode.return_value.decode.return_value,
+            'ips_info': mock_b64encode.return_value.decode.return_value
+        }
+        mock_get_cbslinit_scripts_dir.assert_called_once_with(
+            mock.sentinel.base_dir)
+        mock_write_winrm_file.assert_called_once_with(
+            self.morphing_tools._conn,
+            "%s\\01-static-ip-config.ps1" %
+            mock_get_cbslinit_scripts_dir.return_value,
+            contents.encode())
+
+    @mock.patch.object(
+        windows.BaseWindowsMorphingTools, '_get_cbslinit_base_dir'
+    )
+    @mock.patch.object(
+        windows.BaseWindowsMorphingTools,
+        '_compile_static_ip_conf_from_registry'
+    )
+    @mock.patch.object(
+        windows.BaseWindowsMorphingTools, '_write_static_ip_script'
+    )
+    @mock.patch.object(
+        windows.BaseWindowsMorphingTools, '_unload_registry_hive'
+    )
+    @mock.patch.object(
+        windows.BaseWindowsMorphingTools, '_load_registry_hive'
+    )
+    @mock.patch.object(windows.BaseWindowsMorphingTools, '_check_ips_info')
```

Comment:
Avoid mocking `_check_ips_info` by setting appropriate return value for `_compile_static_ip_conf_from_registry` mock, so that the following code passes:
```python
self._check_ips_info(nics_info, ips_info)
```